src/base/bittorrent/torrenthandle.cpp

```diff
@@ -440,8 +440,15 @@ void TorrentHandle::addTrackers(const QVector<TrackerEntry> &trackers)
     QVector<TrackerEntry> newTrackers;
     newTrackers.reserve(trackers.size());
 
-    for (const TrackerEntry &tracker : trackers) {
+    int nextTier = 0;
+    for (const TrackerEntry &tracker : currentTrackers)
```

Comment:
You should take a copy of `m_nativeHandle.trackers()` at top of this function and loop through it instead of looping `currentTrackers`.

Reply:
Also you should use `std::max_element`:
```c++
const int nextTier = std::max_element(trackers.begin(), trackers.end()
    , [](const lt::announce_entry &left, const lt::announce_entry &right)
    {
        return (left.tier() < right.tier());
    })->tier() + 1;
```

Reply:
We'll still need to create the `currentTrackers` set to determine if the tracker already exists. Is there any difference if we're iterating over that set or iterating over `m_nativeHandle.trackers()`?

Reply:
>Is there any difference if we're iterating over that set or iterating over m_nativeHandle.trackers()?

Yes in efficiency https://stackoverflow.com/questions/16699247/what-is-a-cache-friendly-code
AFAIK `vector` is better than `QSet` in terms of spatial locality.

Reply:
Good point. I've updated it and added protection against the `uint8_t` limit @thalieht mentioned.